receiver/redisreceiver/metadata.yaml

```diff
@@ -8,6 +8,21 @@ resource_attributes:
 attributes:
   state:
     description: Redis CPU usage state
+    # Redis versions < 6.0 have:
+    #   used_cpu_sys: System CPU consumed by the Redis server, which is the sum of system CPU consumed by all threads of the server process (main thread and background threads)
+    #   used_cpu_user: User CPU consumed by the Redis server, which is the sum of user CPU consumed by all threads of the server process (main thread and background threads)
+    #   used_cpu_sys_children: System CPU consumed by the background processes
+    #   used_cpu_user_children: User CPU consumed by the background processes
+    # Redis versions >= 6.0 have two more:
+    #   used_cpu_sys_main_thread: System CPU consumed by the Redis server main thread
+    #   used_cpu_user_main_thread: User CPU consumed by the Redis server main thread
```

Comment:
Hm. Sorry I didn't pay attention to the actual new attributes. According to the specification aggregation over all the attributes must provide meaningful values. Can you please confirm if that was and will be the case?

I'm afraid that `used_cpu_sys_main_thread` is overlapping with `used_cpu_sys`. 

Probably only aggregation over `used_cpu_sys_children` and `used_cpu_sys` (or `used_cpu_user_children` and `used_cpu_user`) provides meaningful values. 

If my assumptions are correct we will need split `redis.cpu.time` into two metrics and add yours as another optional metric. But I want to discuss it first.

Reply:
So used_cpu_sys_main_thread was introduced with Redis 6.0 IO threading. It's not overlapping with used_cpu_sys, but part of used_cpu_sys. used_cpu_sys = used_cpu_main_thread + used_cpu_io_thread. While used_cpu_sys/user_children are the stats for child process.

Reply:
Ok. What sum over all the attributes will represent before and after 6.0?

Reply:
I think we might leave how to aggregate or interpret the values Redis Info cmd gives us to the consumers. OT as collector can focus on relay these values as metrics to consumers.

Reply:
We need to design metrics according to the [OTel spec guidelines](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#general-guidelines) which are saying ["As a rule of thumb, aggregations over all the attributes of a given metric SHOULD be meaningful,"](https://prometheus.io/docs/practices/naming/#metric-names). So I'm trying to understand if we violate this rule or not?

Reply:
That's a valid argument. We definitely can aggregate on different processes like one for main process, one for child process like before 6.0 and as you suggested. But in that case we would lose the information Redis 6.0 gives: how much the main thread used, because now the used_cpu_sys include both the main thread cpu and io thread cpu usage. Additional metrics on used_cpu_sys_main_thread and used_cpu_user_main_thread gives  users that ability to differentiate.